syntax = "proto3";

package crawl;
option go_package = "go/pb";

service Spider {
  rpc CrawlDishesFromGoogleMap (CrawlDishesFromGoogleMapRequest) returns (CrawlDishesFromGoogleMapReply) {}
}

message Dish {
  string name = 1;
  string description = 2;
  string image = 3;
  string price = 4;
  string category = 5;
}

message CrawlDishesFromGoogleMapRequest {
  string googleMapUrl = 1;
}

message CrawlDishesFromGoogleMapReply {
  bool success = 1;
  string message = 2;
  repeated Dish dish = 3;
}